### DISCUSSION TITLE
Saving predictions to file using LightningCLI

### DISCUSSION THREAD

**Discussion by vikigenius** [Lightning-AI/pytorch-lightning#14817] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
The predict command in LightningCLI does not specify anything about saving the predictions to file. Are they saved anywhere?

If not how can I extend LightningCLI pedict command to save the predictions, preferably by saving to a specific location.

I see that there is a [BasePredictionWriter](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.BasePredictionWriter.html#pytorch_lightning.callbacks.BasePredictionWriter) callback, but I have no idea how to use it or add it to the LightningCLI specifically for the predict subcommand

**Comment by tshu-w** [comment]
You can ref https://github.com/Lightning-AI/lightning/discussions/14316. Using the callback is a nice idea in the current situation without hacking the CLI directly. To add a callback to CLI:
```python
LightningCLI(
    ...
    trainer_defaults={
        "callbacks": [
            prediction_callback,
        ]
    },
)
```
or
```console
python cli.py prediction --callbacks+ src.callbacks.prediction_callbacks ...
```
